### PR TITLE
TextInput loading and disabled implemented

### PR DESCRIPTION
**Pull request by dancingshell**
### Summary

* added spinner
* added disabled styles
(text-input is disabled when disabled or loading are true)

> loading 
screen shot 2017-10-18 at 5 26 20 pm
> disabled
screen shot 2017-10-18 at 5 26 00 pm
> default
screen shot 2017-10-18 at 5 26 36 pm

### CODE REVIEW

**Comment by ben-burlingham** [comment]
Perhaps remove the `clear` button while disabled?

File: text-input/src/TextInput.scss [hunk 1]
```diff
@@ -44,21 +44,64 @@ $h: 30px;
...
   }
+}
 
-  &:focus {
-    outline: none;
-  }
+.spinner {
+  border: 2px solid #f3f3f3; /* Light grey */
+  border-top: 2px solid #f47e42; /* Orange */
+  border-radius: 50%;
```
Comment (ben-burlingham) [inline text-input/src/TextInput.scss]: Wow! This is a clean way to solve the spinner, would love to implement the same in Filters 🤔

File: text-input/src/TextInput.scss [hunk 2]
```diff
@@ -44,21 +44,64 @@ $h: 30px;
...
   display: flex;
   justify-content: center;
-  right: $h;
+  height: 100%;
   width: $h;
+  padding: 0;
+  border-radius: 0 1px 1px 0;
 
+  &:focus {
+    outline: none;
```
Comment (mcclayton) [inline text-input/src/TextInput.scss]: `nb`: We keep doing this places for what I assume is aesthetics - but this has a11y implications. Again, I suppose we don't have a big accessibility push, but in the future accessibility can drive who we can and can't sell our product to.

File: text-input/src/TextInput.scss (hunk 1 above)
Comment (dancingshell) [reply text-input/src/TextInput.scss]: Compliments of @nickwmurphy / the internet

File: text-input/test/TextInput.test.jsx [hunk 3]
```diff
@@ -93,10 +93,53 @@ describe('TextInput', () => {
     });
 
     describe('When disabled is true', () => {
+      const props = { disabled: true, onChange: jest.fn(), scope: 'MOCK_SCOPE' };
```
Comment (mcclayton) [inline text-input/test/TextInput.test.jsx]: Does onChange need to be a Jest spy here? Doesn't look like it's used anywhere if I am seeing correctly.

File: text-input/test/TextInput.test.jsx [hunk 4]
```diff
@@ -93,10 +93,53 @@ describe('TextInput', () => {
...
+      const disabledTextInput = renderTextInput(props, mount);
```
Comment (mcclayton) [inline text-input/test/TextInput.test.jsx]: I don't think we are testing anything that needs React lifecycle methods here, so we don't need to use Enzyme's http://airbnb.io/enzyme/docs/api/mount.html
Can just shallow render. (Omit the renderer param of `renderTextInput`

File: text-input/test/TextInput.test.jsx [hunk 5]
```diff
@@ -93,10 +93,53 @@ describe('TextInput', () => {
...
+        const spinner = loadingTextInput.find('.clear');
+
+        it('should not be shown', () => {
+          expect(spinner.exists()).toBe(false);
+        });
+
+        describe('When clicked', () => {
+          it('should not call onChange', () => {
+            loadingTextInput.find('button').simulate('click');
+            // Should be called with empty value and scope
```
Comment (mcclayton) [inline text-input/test/TextInput.test.jsx]: Comment needs updated or removed.

File: text-input/test/TextInput.test.jsx [hunk 6]
```diff
@@ -93,10 +93,53 @@ describe('TextInput', () => {
...
+    describe('When loading is true', () => {
+      const props = { loading: true, onChange: jest.fn(), scope: 'MOCK_SCOPE' };
+      const loadingTextInput = renderTextInput(props, mount);
+
+      it('should be disabled', () => {
+        expect(loadingTextInput.find('input').prop('disabled')).toBe(true);
+      });
+
+      describe('Clear Button', () => {
+        const spinner = loadingTextInput.find('.clear');
```
Comment (mcclayton) [inline text-input/test/TextInput.test.jsx]: Variable name is spinner, but we are selecting the clear icon?

File: text-input/test/TextInput.test.jsx (hunk 4 above)
Comment (mcclayton) [reply text-input/test/TextInput.test.jsx]: http://airbnb.io/enzyme/docs/api/shallow.html

**Pull request merged**